Black-Market/Black-Market/Components/MarketButton.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  MarketButton.swift
+//  Black-Market
+//
+//  Created by Junior Sancho on 7/10/23.
+//
+
+import SwiftUI
+
+struct MarketButton: View {
+    @Binding var style: Style
+    private let action: () -> Void
+    private let title: String
+        
+    enum Style {
+        case filled
+        case disabled
+        case plain
+    }
+    
+    init(
+        style: Binding<Style> = .constant(.plain),
+        action: @escaping () -> Void,
+        title: String
+    ) {
+        self._style = style
+        self.action = action
+        self.title = title
+    }
+
+    var body: some View {
+        Button {
+            action()
+        } label: {
+            Text(title)
+                .applyButtonStyle(style)
+        }
+        .disabled(style == .disabled)
+    }
+}
+
+struct FilledButtonStyle: ViewModifier {
+    
+    func body(content: Content) -> some View {
+        content
+            .frame(maxWidth: .infinity, maxHeight: 44)
+            .background(Color("BlackStyle"))
+            .foregroundColor(.white)
+            .cornerRadius(8)
+    }
+}
+
+struct PlainButtonStyle: ViewModifier {
+    
+    func body(content: Content) -> some View {
+        content
+            .frame(maxWidth: .infinity, maxHeight: 44)
+            .background(.clear)
+            .foregroundColor(Color("BlackStyle"))
```

Comment:
It would be nice to have all the Colors in a separate file (Color extension) and invoke them from there. This way, you wouldn't have to duplicate the same string across multiple screens when using an color, ie:

```
extension Color {
  static let black = Color("BlackStyle")
  static let lightGrey = Color("LightGreyStyle")
}
```

Reply:
Done